src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/AnnotationTransactionMapper.java

```diff
@@ -51,9 +61,19 @@ public FilingHistoryDocument mapFilingHistoryToExistingDocumentUnlessStale(Inter
                                             annotationChildMapper.mapChild(annotation, request);
                                         },
                                         // Add new annotation to existing annotations list
-                                        () -> annotationList
-                                                .add(annotationChildMapper
-                                                        .mapChild(new FilingHistoryAnnotation(), request))),
+                                        () -> {
+                                            if (annotationList.stream()
+                                                    .anyMatch(annotation ->
+                                                            StringUtils.isBlank(annotation.getEntityId()))) {
+                                                LOGGER.error(
```

Comment:
Is this really an error? It will be a normal event to annotate a filing history document created by the Perl code, so can this log message be changed to `info`? Errors should really be reserved for something that Fess should react to when we go Live.